middleware/redirects.global.ts

```diff
@@ -15,36 +16,47 @@ export default defineNuxtRouteMiddleware((route) => {
     }
   }
 
-  let redirectValue
+  let redirectValue: RouteLocationRaw | null | undefined
 
   const paths = [
     {
-      cond: val => val === '/drops',
+      cond: (val: string) => val === '/drops',
       replaceValue: '/ahp/drops',
     },
     {
-      cond: val =>
+      cond: (val: string) =>
+        val === `/${urlPrefix.value}/profile`,
+      replaceValue: () => {
+        const { accountId } = useAuth()
+        if (accountId.value) {
+          return `/${urlPrefix.value}/u/${accountId.value}`
+        }
+        return `/${urlPrefix.value}`
```

Comment:
nit: since it's only one if else 
```suggestion
        return accountId.value ? `/${urlPrefix.value}/u/${accountId.value}` : `/${urlPrefix.value}`
```